### DISCUSSION TITLE
Setup Cypress component testing in an Angular library project

### DISCUSSION THREAD

**Discussion by robbannn** [cypress-io/cypress#29661] [Questions and Help]
I want to know how to set up Cypress component testing in an Angular library project.

[This](https://github.com/cypress-io/cypress/issues/23677) issue is saying that you need to have an application in your project. Is this still necessary?
If it is, how are we supposed to set that up?

I would really appreciate som help and guidance :) 

I'm running:

Angular 18.0.1 and Cypress 13.11.0

**Comment by portyu9** [comment]
Hi @robbannn 

For current Cypress, you do **not** need to create a dummy Angular application solely so Cypress can discover build configuration.

Angular Component Testing now supports providing the build context explicitly through:

`component.devServer.options.projectConfig`

That lets you supply your library's `sourceRoot`, `tsConfig`, styles/assets, and other required Angular build options instead of Cypress looking for a default `application` project in `angular.json`.

The important version distinction is that this thread was using Cypress `13.11.0` with Angular `18.0.1`, where the older application-detection limitation could still apply.

If setting this up today, use the `projectConfig` approach. Also note that current Cypress 16 Component Testing supports Angular 21 and 22, so an Angular 18 project would need to stay on an older compatible Cypress version or upgrade Angular.

Please mark this answer as accepted if it helped, thank you.
